openapi/openapi/src/main/scala/endpoints/openapi/model/OpenApi.scala

```diff
@@ -306,14 +339,34 @@ object OpenApi {
         mutable.LinkedHashMap(
           "openapi" -> ujson.Str(openApiVersion),
           "info" -> infoJson(openApi.info),
-          "paths" -> pathsJson(openApi.paths)
+          "paths" -> pathsJson(openApi.paths),
         )
+      if (openApi.tags.nonEmpty) {
+       fields += "tags" -> ujson.Arr(openApi.tags.map(tagJson): _*)
+      }
       if (openApi.components.schemas.nonEmpty || openApi.components.securitySchemes.nonEmpty) {
         fields += "components" -> componentsJson(openApi.components)
       }
       new ujson.Obj(fields)
     }
 
+  private def extractTags(paths: Map[String, PathItem]): List[Tag] = {
+    val allTags = paths.flatMap { case (_, pathItem) =>
+      pathItem.operations.map { case (_, operation) =>
+        operation.tags
+      }
+    }.flatten
+
+    val tagsByName = allTags.groupBy(_.name)
+    tagsByName.foreach { case (_, listOfTags) =>
+      val set = listOfTags.toSet
+      if (set.size > 1) {
+        throw new IllegalArgumentException(s"Found tags with the same name but different values: $set")
+      }
+    }
+    allTags.filter(tag => tag.description.nonEmpty || tag.externalDocs.nonEmpty).toSet.toList
```

Comment:
Why do we keep only those whose description or external doc is defined?

Reply:
The tags at the root are used only when there is more information in the tag than just the name. So, if a tag is just a name there is no value in adding it in the root anyway. It will still be displayed as before.

Reply:
OK, thanks for the clarification! Do we have a test for that? It seems that in `ReferencedSchemaTest` we only use a “rich” tag?

Reply:
Sure, I'll add another case t the `ReferencedSchemaTest`.